src/components/svgicon/SVGIcon.tsx

```diff
@@ -0,0 +1,45 @@
+import React from "react";
+import styled, {css} from "styled-components";
+import theme from "../../theme";
+import {getViewBox} from "../../utils/svgicons";
+import {getPath} from "../../utils/svgicons";
+
+export interface SVGIconProps {
+    name: string;
+    fill?: "success" | "danger" | "warning" | "notification" | "white",
+    size?: "sm" | "md" | "lg" | "xl" | "xxl"
+}
+
+const StyledSVGIcon = styled.svg<SVGIconProps>`
+    fill: ${theme.colors.primary};
+    width: ${theme.iconSize.md};
+    height: ${theme.iconSize.md};
+    
+    ${({fill}) =>
+    fill && css`
+        fill: ${theme.colors[fill]};
+    `}
+    
+    ${({size}) =>
+    size && css`
+        width: ${theme.iconSize[size]};
+        height: ${theme.iconSize[size]};
+    `}
+`;
+
+export const SVGIcon: React.FC<SVGIconProps> = (props) => {
```

Comment:
You can use "destructuring" to get "name" from "props".

Reply:
`export const SVGIcon: React.FC<SVGIconProps> = ({ name, ...props }) => {`

and this to SVGIconProps 
`[x: string]: any;`

Reply:
done